src/Aspire.Hosting.Azure/Provisioning/Provisioners/AzureProvisioner.cs

```diff
@@ -291,13 +291,13 @@ private async Task ProcessResourceAsync(IConfiguration configuration, Lazy<Task<
 
             resourceLogger.LogWarning("No provisioner found for {resourceType} skipping.", resource.GetType().Name);
         }
-        else if (!provisioner.ShouldProvision(configuration, resource.AzureResource))
+        else if (!resource.AzureResource.IsExisting() && !provisioner.ShouldProvision(configuration, resource.AzureResource))
```

Comment:
Why do you need these checks?

Reply:
To avoid provisioning a resource or reading the config that has been cached in user-secrets when the same config name is stored. Without this, starting with:

```csharp
var bus = builder.AddServiceBus("messages");
```

And moving to:

```csharp
var bus = builder.AddServiceBus("messages").RunAsExisting("cached-messaging");
```

would still load the config for the Aspire-managed "messages" resource from user-secrets.

Reply:
Shouldn't that just work? I'm confused what this done.